docs/framework/interop/default-marshaling-for-strings.md

```diff
@@ -1,264 +1,281 @@
+
+Strings are marshaled as a COM-style `BSTR` type or as a null-terminated string (a character array that ends with a null character). The characters within the string can be marshaled as Unicode (the default on Windows systems) or ANSI.
+
+## Strings Used in Interfaces
+
+The following table shows the marshaling options for the string data type when marshaled as a method argument to unmanaged code. The <xref:System.Runtime.InteropServices.MarshalAsAttribute> attribute provides several <xref:System.Runtime.InteropServices.UnmanagedType> enumeration values to marshal strings to COM interfaces.
+
+|Enumeration type|Description of unmanaged format|
+|----------------------|-------------------------------------|
+|`UnmanagedType.BStr` (default)|A COM-style `BSTR` with a prefixed length and Unicode characters.|
+|`UnmanagedType.LPStr`|A pointer to a null-terminated array of ANSI characters.|
+|`UnmanagedType.LPWStr`|A pointer to a null-terminated array of Unicode characters.|
+
+This table applies to <xref:System.String>. For <xref:System.Text.StringBuilder>, the only options allowed are `UnmanagedType.LPStr` and `UnmanagedType.LPWStr`.
+
+The following example shows strings declared in the `IStringWorker` interface.
+
+```csharp
+public interface IStringWorker
+{
+    void PassString1(string s);
+    void PassString2([MarshalAs(UnmanagedType.BStr)] string s);
+    void PassString3([MarshalAs(UnmanagedType.LPStr)] string s);
+    void PassString4([MarshalAs(UnmanagedType.LPWStr)] string s);
+    void PassStringRef1(ref string s);
+    void PassStringRef2([MarshalAs(UnmanagedType.BStr)] ref string s);
+    void PassStringRef3([MarshalAs(UnmanagedType.LPStr)] ref string s);
+    void PassStringRef4([MarshalAs(UnmanagedType.LPWStr)] ref string s);
+}
+```
+
+```vb
+Public Interface IStringWorker
+    Function PassString1(s As String)
+    Function PassString2(<MarshalAs(UnmanagedType.BStr)> s As String)
+    Function PassString3(<MarshalAs(UnmanagedType.LPStr)> s As String)
+    Function PassString4(<MarshalAs(UnmanagedType.LPWStr)> s As String)
+    Function PassStringRef1(ByRef s As String)
+    Function PassStringRef2(<MarshalAs(UnmanagedType.BStr)> ByRef s As String)
+    Function PassStringRef3(<MarshalAs(UnmanagedType.LPStr)> ByRef s As String)
+    Function PassStringRef4(<MarshalAs(UnmanagedType.LPWStr)> ByRef s As String)
+End Interface
 ```
 
 The following example shows the corresponding interface described in a type library.
 
+```cpp
+[...]
+interface IStringWorker : IDispatch
+{
+    HRESULT PassString1([in] BSTR s);
+    HRESULT PassString2([in] BSTR s);
+    HRESULT PassString3([in] LPStr s);
+    HRESULT PassString4([in] LPWStr s);
+    HRESULT PassStringRef1([in, out] BSTR *s);
+    HRESULT PassStringRef2([in, out] BSTR *s);
+    HRESULT PassStringRef3([in, out] LPStr *s);
+    HRESULT PassStringRef4([in, out] LPWStr *s);
+};
 ```
-[…]  
```

Comment:
Is this useful?

Reply:
No, I don't think it's useful, @NextTurn. I have no idea of why it's there.